zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -143,30 +143,61 @@ object Cookie {
    * Decodes from Set-Cookie header value inside of Response into a cookie
    */
   def decodeResponseCookie(headerValue: String): Option[Cookie] = {
-    val cookieWithoutMeta = headerValue.split(";").map(_.trim)
-    val (first, other)    = (cookieWithoutMeta.head, cookieWithoutMeta.tail)
-    val (name, content)   = splitNameContent(first)
-
-    val cookie =
-      if (name.trim == "" && content.isEmpty) Option.empty[Cookie]
-      else Some(Cookie(name, content.getOrElse("")))
-
-    other.map(splitNameContent).map(t => (t._1.toLowerCase, t._2)).foldLeft(cookie) {
-      case (Some(c), ("expires", Some(v)))  => parseDate(v).map(c.withExpiry(_))
-      case (Some(c), ("max-age", Some(v)))  => Try(v.toLong).toOption.map(c.withMaxAge(_))
-      case (Some(c), ("domain", v))         => Some(c.withDomain(v.getOrElse("")))
-      case (Some(c), ("path", v))           => Some(c.withPath(Path(v.getOrElse(""))))
-      case (Some(c), ("secure", _))         => Some(c.withSecure)
-      case (Some(c), ("httponly", _))       => Some(c.withHttpOnly)
-      case (Some(c), ("samesite", Some(v))) =>
-        v.trim.toLowerCase match {
-          case "lax"    => Some(c.withSameSite(SameSite.Lax))
-          case "strict" => Some(c.withSameSite(SameSite.Strict))
-          case "none"   => Some(c.withSameSite(SameSite.None))
-          case _        => Some(c)
-        }
-      case (c, _)                           => c
+    var name: String              = null
+    var content: String           = null
+    var expires: Instant          = null
+    var maxAge: Option[Long]      = None
+    var domain: String            = null
+    var path: Path                = null
+    var secure: Boolean           = false
+    var httpOnly: Boolean         = false
+    var sameSite: Cookie.SameSite = null
+
+    var prev = 0
+    var i    = 0
+    while (i >= 0) {
+      i = headerValue.indexOf(';', prev)
+      val c       = if (i <= 0) headerValue.substring(prev) else headerValue.substring(prev, i)
+      val (n, ct) = splitNameContent(c)
+
+      (n.toLowerCase, ct) match {
```

Comment:
unnecessary tuple allocation.

Reply:
@tusharmath I have mode some changes to avoid string and tuple unnecessary allocations, inner loop is not the fanciest but I think that this way you get more performance ... what do you think ?

Reply:
Can you commit the `Benchmark` code to the benchmarks project.

Reply:
Sure, added CookieDecodeBenchark to zio-http-benchmarks (very naive at the moment)